aiida_restapi/routers/nodes.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+"""Declaration of FastAPI application."""
+
+try:
+    from importlib import metadata
+except ImportError:  # for Python<3.8
+    import importlib_metadata as metadata  # type: ignore[no-redef]
+
+from aiida.cmdline.utils.decorators import with_dbenv
+from fastapi import APIRouter, Depends, File
+
+from aiida_restapi.models import Node, User
+
+from .auth import get_current_active_user
+
+router = APIRouter()
+
+
+@router.post("/nodes", response_model=Node)
+@with_dbenv()
+async def create_node(
+    node: Node,
+    current_user: User = Depends(
+        get_current_active_user
+    ),  # pylint: disable=unused-argument
+) -> Node:
+    """Create new AiiDA node."""
+
+    node_dict = node.dict(exclude_unset=True)
+    node_type = node_dict.pop("node_type", None)
+    attributes = node_dict.pop("attributes", None)
```

Comment:
won't the extra be set when passing parameter using **node_dict?

Reply:
I see, maybe - can you please add a test?

If this works fine, since you aren't doing anything specific with the attributes here besides passing them along, perhaps it would be better if the attributes would be popped out from the nodes_dict inside `create_new_node` instead?

Reply:
@ltalirz  You were correct about extras, I have added extras and test for extras.